### PR TITLE
fix(TDI-46675): fix for Couchbase schema widget in PD.

### PR DESCRIPTION
**Pull request by dmytrogrygorenko**
https://jira.talendforge.org/browse/TDI-46675
Fix for schema widget for Couchbase in PD.

### CODE REVIEW

File: couchbase/src/main/java/org/talend/components/couchbase/dataset/CouchbaseDataSet.java [hunk 1]
```diff
@@ -36,21 +39,22 @@
...
+    @Documentation("Connection.")
     private CouchbaseDataStore datastore;
 
     @Option
-    @Documentation("Schema")
-    @Structure(type = Structure.Type.OUT, discoverSchema = "discover")
-    private List<String> schema;
+    @Documentation("Schema.")
+    @Structure(type = Structure.Type.IN, discoverSchema = "discover")
+    @Suggestable(value = "listColumns", parameters = { "datastore", "bucket" })
```
Comment (mbasiuk-talend) [inline couchbase/src/main/java/org/talend/components/couchbase/dataset/CouchbaseDataSet.java:48]: it's better to use constants in suggestable names. Could you please extract it.

File: couchbase/src/main/java/org/talend/components/couchbase/service/CouchbaseService.java [hunk 2]
```diff
@@ -172,6 +175,24 @@ public Schema addColumns(@Option("dataSet") final CouchbaseDataSet dataSet) {
         return record.getSchema();
     }
 
+    @Suggestions("listColumns")
+    public SuggestionValues listColumns(@Option("datastore") final CouchbaseDataStore datastore,
+            @Option("bucket") String bucketName) {
+        List<SuggestionValues.Item> items = new ArrayList<>();
```
Comment (mbasiuk-talend) [inline couchbase/src/main/java/org/talend/components/couchbase/service/CouchbaseService.java:181]: you may not create an instance here. It's easier in a stream termination call step.

File: couchbase/src/main/java/org/talend/components/couchbase/service/CouchbaseService.java [hunk 3]
```diff
@@ -172,6 +175,24 @@ public Schema addColumns(@Option("dataSet") final CouchbaseDataSet dataSet) {
...
+        CouchbaseDataSet dataset = new CouchbaseDataSet();
+        dataset.setDatastore(datastore);
+        dataset.setBucket(bucketName);
+        try {
+            Schema schema = addColumns(dataset);
+            List<Schema.Entry> entries = schema.getEntries();
+            entries.forEach(e -> items.add(new SuggestionValues.Item(e.getName(), e.getName())));
+            return new SuggestionValues(true, items);
+        } catch (Exception e) {
+            LOG.error(e.getMessage());
```
Comment (mbasiuk-talend) [inline couchbase/src/main/java/org/talend/components/couchbase/service/CouchbaseService.java:191]: Is it correct, if we catch an Exception we would *cache* the empty values? next line after this block.

File: couchbase/src/main/resources/org/talend/components/couchbase/Messages.properties [hunk 4]
```diff
@@ -4,4 +4,5 @@ Couchbase.dataset.CouchbaseDataSet._placeholder=
 Couchbase.datastore.CouchbaseDataStore._displayName=Couchbase
 Couchbase.datastore.CouchbaseDataStore._placeholder=
 Couchbase.actions.healthcheck.healthCheck._displayName=Check connection
-Couchbase.actions.schema.discover._displayName=Discover schema
\ No newline at end of file
+Couchbase.actions.schema.discover._displayName=Discover schema
+Couchbase.actions.suggestions.listColumns._displayName=
```
Comment (mbasiuk-talend) [inline couchbase/src/main/resources/org/talend/components/couchbase/Messages.properties:8]: Have you tried to name it?

**Review by mbasiuk-talend — CHANGES_REQUESTED** [review]

File: couchbase/src/main/java/org/talend/components/couchbase/dataset/CouchbaseDataSet.java (hunk 1 above)
Comment (dmytrogrygorenko) [reply couchbase/src/main/java/org/talend/components/couchbase/dataset/CouchbaseDataSet.java]: Fixed. Extracted to constants.

File: couchbase/src/main/java/org/talend/components/couchbase/service/CouchbaseService.java (hunk 3 above)
Comment (dmytrogrygorenko) [reply couchbase/src/main/java/org/talend/components/couchbase/service/CouchbaseService.java:195]: Fixed.

File: couchbase/src/main/resources/org/talend/components/couchbase/Messages.properties (hunk 4 above)
Comment (dmytrogrygorenko) [reply couchbase/src/main/resources/org/talend/components/couchbase/Messages.properties]: Name assigned to label.

**Pull request merged**